ring-jetty-adapter/src/ring/adapter/jetty.clj

```diff
@@ -28,18 +28,48 @@
           (servlet/update-servlet-response response response-map)
           (.setHandled base-request true))))))
 
-(defn- ^AbstractHandler async-proxy-handler [handler timeout]
+(defn- async-jetty-raise
+  [^AsyncContext context ^HttpServletResponse response]
+  (fn [^Throwable exception]
+    (.sendError ^HttpServletResponse response
+                500
+                (.getMessage exception))
+    (.complete ^AsyncContext context)))
+
+(defn- async-jetty-respond
```

Comment:
This could be more succinctly written as:

```clojure
(defn- async-jetty-respond [context response]
  (fn [response-map]
    (servlet/update-servlet-response response context response-map)))
```